### PR TITLE
#2061 When all QAs on vacation in the project, notify ARCs about that

### PR DESCRIPTION
**Pull request by sbzDev**
https://github.com/zerocracy/farm/issues/2061

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#2063](https://github.com/zerocracy/farm/issues/2063) is now in scope, role is `REV`

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy [hunk 1]
```diff
@@ -46,8 +46,24 @@ def exec(Project project, XML xml) {
   Cash cash = new Cash.S(claim.param('cash'))
   Cash bonus = new Cash.S(claim.param('bonus'))
   People people = new People(farm).bootstrap()
-  List<String> qaList = new Roles(project).bootstrap().findByRole('QA')
+  Roles roles = new Roles(project).bootstrap()
+  List<String> qaList = roles.findByRole('QA')
   Collection<String> qa =  new Filtered<String>({ uid -> !people.vacation(uid) }, qaList)
+  if (qa.empty){
+    roles.findByRole('ARC').each { arc ->
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: @sbzDev there's no need to notify all arcs, lets ping primary ARC only: `String arc = roles.findByRole('ARC')[0]` (like in all other places where ARC is notified).

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy [hunk 2]
```diff
@@ -46,8 +46,24 @@ def exec(Project project, XML xml) {
...
+      claim.copy()
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: @sbzDev let's use `claim.reply()` to notify ARC right in the ticket, it automatically creates a token for GitHub ticket:
```java
claim.reply("some message");
```

**Comment by sbzDev** [comment]
@g4s8 please check this PR

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy [hunk 3]
```diff
@@ -46,8 +46,24 @@ def exec(Project project, XML xml) {
...
+        .type('Notify user')
+        .token("user;$arc")
+        .param(
+          'message',
+          new Par(
+            farm,
+            'All QAs in the project %s are on vacation. Please handle that.'
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: @sbzDev if we use `claim.reply` we don't need to know project id here, but need to mention ARC by GitHub username: "@%s all QAs..."

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy (hunk 2 above)
Comment (sbzDev) [reply src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: fixed

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy (hunk 1 above)
Comment (sbzDev) [reply src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: fixed

File: src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy (hunk 3 above)
Comment (sbzDev) [reply src/main/groovy/com/zerocracy/stk/pm/qa/start_qa_review.groovy]: fixed

**Comment by g4s8** [comment]
@sbzDev thanks

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#2063` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#2063](https://github.com/zerocracy/farm/issues/2063) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**